### ISSUE TITLE
Stretched font (SourceCodePro)

### ISSUE THREAD

**Issue by tombh** [wez/wezterm#2035]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### WezTerm version

wezterm 20220525-074534-385f3c99

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Wezterm:
![image](https://user-images.githubusercontent.com/160835/170337758-1a0d81ff-04df-4378-84f1-39b29a8a9d3c.png)
Alacritty:
![image](https://user-images.githubusercontent.com/160835/170337823-1d107587-c1b5-4084-8986-1496d4b7a568.png)

Mainly notice how Alacritty honours the 9-point blockness of the font. I assume whatever causes the difference between Alacritty and Wezterm also causes a slight stretching of the main font itself.

### To Reproduce

See my pasted config.

### Configuration

```lua
local wezterm = require("wezterm")
return {
	enable_wayland = true,
	window_decorations = "NONE",
	enable_tab_bar = false,
	window_padding = {
		top = 0,
		bottom = 0,
		left = 0,
		right = 0,
	},
	font = wezterm.font_with_fallback({ "SourceCodePro", "SauceCodePro Nerd Font", "DejaVu Sans" }),
	font_size = 11.2,
	bold_brightens_ansi_colors = true,
	custom_block_glyphs = false,
	color_scheme = "Floraverse",
}
```

I'm using "SourceCodePro" and "SauceCodePro Nerd Font" because, as has been mentioned elsewhere in the issues, I indeed found it true that patched fonts don't render so well (although it's not a problem on Alacritty). For example, notice the truncated "1" when "SauceCodePro Nerd Font" is made the primary font:
![image](https://user-images.githubusercontent.com/160835/170338949-2258caca-359d-434f-b766-27d5f0da4971.png)

Also noteworthy is that I'm using "DejaVu Sans" to get a better (bit not quite perfect braille font), here's a screenshot without "DejaVu Sans" in the list: 
![image](https://user-images.githubusercontent.com/160835/170340013-59220327-bb5e-4757-9999-625798747e9a.png)

### Expected Behavior

See the Alacritty screenshot above.

### Logs

15:16:12.733  INFO   wezterm_gui::termwindow > OpenGL initialized! Mesa Intel(R) Iris(R) Plus Graphics (ICL GT2) 4.6 (Compatibility Profile) Mesa 22.0.4 is_context_loss_possible=false wezterm version: 20220525-074534-385f3c99
15:16:12.896  WARN   wezterm_term::terminalstate > unhandled XtermKeyMode OtherKeys Some(1)
15:16:17.386  WARN   wezterm_term::terminalstate > unhandled DecPrivateMode ResetDecPrivateMode(Unspecified(1005))
15:16:17.398  WARN   wezterm_term::terminalstate > unhandled DecPrivateMode SetDecPrivateMode(Unspecified(7727))
15:16:17.398  WARN   wezterm_term::terminalstate > unhandled DecPrivateMode SetDecPrivateMode(Unspecified(7727))

### Anything else?

Yes! A little anecdote as to why I came to try Wezterm. So I've got a pet project, it's Rust-based terminal multiplexing stuff. At first I was just getting PTY behaviour from the classic `script` command and emulating ANSI behaviour with https://github.com/doy/vt100-rust. Then a few days ago, I just happened to come across [a Reddit post](https://www.reddit.com/r/linux/comments/uvsixj/mprocs_022_tui_for_running_multiple_processes_in/) where somebody had done some cool multiplexing in Rust too, I peeked they're [`Cargo.toml`](https://github.com/pvolok/mprocs/blob/master/Cargo.toml#L34) for goodies, and saw `portable-pty` 😏 Nice, I thought, that's better than calling out to the shell. But still not quite satisfied with the somewhat stagnating vt100 crate, I suddenly remembered [Zellij](https://github.com/zellij-org/zellij) and wondered how _they_ were embedding terminals, and that's when I found `termwiz` 😏😏 (I still haven't tried it yet, but I think I can make an in-memory ANSI terminal right?).

Anyway, seeing both those cool projects as kindly-refactored crates for the community to enjoy, I thought, well it's my duty to support Wezterm's generosity by using it. Hence why I'm here now 😊

Edit: I forgot the most important part of the anecdote! The clincher was this comment in the docs:
```rust
/// Specify whether you want to slowly or rapidly annoy your users
// [...]
pub enum Blink {
    None = 0,
    Slow = 1,
    Rapid = 2,
}
```
I'm just not used to belly laughing from reading through documentation 🤣

**Comment by wez** [comment]
Thanks for sharing your anecdote!  I'd forgotten about that comment, and it made me laugh reading it again just now as well!

**Comment by wez** [comment]
Running `wezterm ls-fonts` and, separately, `wezterm ls-fonts --text "sample"` can sometimes be helpful to understand more about what is happening with font selection.  Sometimes the requested font size can influence which font is used as well.

However, that kind of misalignment/stretching suggests a possible *hinting* issue.  Hinting is a process in glyph rasterization that tries to make the vector strokes line up to pixel boundaries so that the glyph looks better on the screen; for example, a vertical line looks better when it fits directly inside a pixel rather than on the line between two pixels.

Font size, dpi and display scaling can influence/exacerbate hinting issues. I noticed that you have selected a fractional point size (presumably to visually match what you previously had in alacritty). That is a bit more likely to lead to this sort of issue.

wezterm has a couple of knobs that can help you dial in how freetype rasterizes:

* https://wezfurlong.org/wezterm/config/lua/config/freetype_load_flags.html
* https://wezfurlong.org/wezterm/config/lua/config/freetype_render_target.html

**Comment by tombh** [comment]
Thanks for those tips, I found what I needed: `freetype_load_flags = "NO_HINTING"`, everything's pretty much almost perfect now. There's no clipping of glyphs when "SauceCodePro Nerd Font" is the primary font (I prefer using my own patched font because its Nerd Fonts are just slightly smaller then Wezterm's). And also I don't need "DejaVu Sans" for braille because those glyphs are rendering well now (though their position is quite different, but that's not an issue, just interesting). I did need a rounded font size, but that was just to ensure those `>` glyphs in nvim/tmux status lines line up. 

So my key stanzas are:
```lua
font = wezterm.font_with_fallback({ "SauceCodePro Nerd Font" }),
freetype_load_flags = "NO_HINTING",
font_size = 11.0,
custom_block_glyphs = false,
```

Thank you 😊
